heron/spi/src/java/com/twitter/heron/spi/statemgr/SchedulerStateManagerAdaptor.java

```diff
@@ -122,7 +122,9 @@ public Boolean setSchedulerLocation(
    * @return Boolean - Success or Failure
    */
   public Boolean deleteTMasterLocation(String topologyName) {
-    return awaitResult(delegate.deleteTMasterLocation(topologyName));
+    return awaitResult(delegate.tmasterLocationExists(topologyName))
+        ? awaitResult(delegate.deleteTMasterLocation(topologyName))
+        : true;
```

Comment:
This style of implementation could lead to one more remote call. Could we figure out a more-efficient way?
Similar to below.